src/corehost/cli/fxr/fx_muxer.cpp

```diff
@@ -1234,6 +1437,85 @@ int fx_muxer_t::execute(
     return result;
 }
 
+int fx_muxer_t::get_coreclr_delegate(
+    const host_startup_info_t &host_info,
+    coreclr_delegate_type type,
+    void **delegate)
+{
+    assert(host_info.is_valid());
+
+    const host_mode_t mode = host_mode_t::libhost;
+
+    // Read config
+    fx_definition_vector_t fx_definitions;
+    auto app = new fx_definition_t();
+    fx_definitions.push_back(std::unique_ptr<fx_definition_t>(app));
+
+    pal::string_t runtime_config = _X("");
+    fx_reference_t override_settings;
+    int rc = read_config(*app, host_info.app_path, runtime_config, override_settings);
+    if (rc != StatusCode::Success)
+        return rc;
+
+    runtime_config_t app_config = app->get_runtime_config();
+    bool is_framework_dependent = app_config.get_is_framework_dependent();
+    if (is_framework_dependent)
+    {
+        fx_name_to_fx_reference_map_t newest_references;
```

Comment:
If possible, I would like to see this 'common' path be resolved using the common path and not here.  My concern is that we have multiple places to keep in sync and drift is likely.

Reply:
As @jeffschwMSFT mentions, it would be nice to refactor this into a separate method to make it easier to reuse as part of the IJW hosting work. I can do that in the IJW PR though after this is merged in.

Reply:
I can do that, but honestly, the last 3 lines and signature are all that is special. IJW will be similar.

Reply:
Sounds good. I'll handle this as part of the IJW work.